src/worker.js

```diff
@@ -280,7 +280,6 @@ self.onmessage = function(e) {
       if (Module['_pthread_self']()) {
         Module['__emscripten_thread_exit'](-1/*PTHREAD_CANCELED*/);
       }
-      postMessage({ 'cmd': 'cancelDone' });
```

Comment:
I wonder if the removal on `cancelDone` can be split out into its own change?   It seems like a bugfix its own right as `__emscripten_thread_exit` should always be enough to recycle the thread.    That would be very small bugix PR that would simplify this one.

Reply:
Reverted with commit 3008b16e7fd210aa1cc192bb2f4f77de43eb9b4c, moved to PR #15624.